src/main/java/org/apache/ibatis/migration/options/SelectedOptions.java

```diff
@@ -15,6 +15,7 @@
  */
 package org.apache.ibatis.migration.options;
 
+// This could just be a Set<Option> selectedOptions = new EnumSet<Option>(Option.class);
```

Comment:
If Options were a binary thing then yes but unfortunately they are not. Although having them be strategies to get the option value might be interesting.

Reply:
can I have and option more than once? Then a `Set` is perfect.

This class is literally a collection of properties. The properties are either: set (1) or not set (0). That's the definition of binary. And before you ask: `null` is a set value of `null`

Imagine an option for each of these properties and each option were a consumer

```java
Option quiet = new QuietOption();
Set<Option> options = new HashSet<>();
options.add(quiet);
quiet.apply(thisApp);

```

```java
public class QuietOption implements Option {

   public void apply(ThisApp app) {
      app.setOutputStream(new OutputStream() { void write(byte thing){}});
   }
}
```

As you can see: With the other changes I made you now have to APPLY the options separately. The data and the functionality that applies to it should go hand-in-hand ala OOP

Reply:
Your second comment is exactly what I said about options being strategies. Sorry if that wasn't clear. It's also shows what I meant about options not being binary things. The option itself yes, the values they potentially represent no.

Reply:
All that means is it's a `List<Option>` then. But if you decide to go with a library for this then you'll end up doing whatever the library says